File: benchmark/nodejs/sieve.js
const limit = 10000;
const sieve = new Array(limit + 1).fill(true);
sieve[0] = sieve[1] = false;
const start = Date.now();
for (let p = 2; p * p <= limit; p++) {
    if (sieve[p]) {
        for (let m = p * p; m <= limit; m += p) {
            sieve[m] = false;
        }
    }
}
let count = 0;
for (let i = 2; i <= limit; i++) {
    if (sieve[i]) count++;
}
const end = Date.now();
console.log(`Primes up to 10000: ${count}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/ruby/sieve.rb
limit = 10000
sieve = Array.new(limit + 1, true)
sieve[0] = sieve[1] = false
start = Time.now.to_f * 1000
(2..Math.sqrt(limit).to_i).each do |p|
  if sieve[p]
    (p*p..limit).step(p) { |m| sieve[m] = false }
  end
end
count = 0
sieve.each { |x| count += 1 if x }
finish = Time.now.to_f * 1000
puts "Primes up to 10000: #{count}"
puts "Time: #{(finish - start).round}ms"

